katello/hooks/boot/01-helpers.rb

```diff
@@ -47,11 +52,15 @@ def execute_command(command, do_say, do_log)
   end
 end
 
-# FIXME: remove when #23332 is released
 module HookContextExtension
+  # FIXME: remove when #23332 is released
   def param_value(mod, name)
     param(mod, name).value if param(mod, name)
   end
+
+  def local_postgresql?
+    param_value('foreman', 'db_manage') || param_value('katello', 'candlepin_manage_db')
+  end
```

Comment:
How about adding the `fail_and_exit` this way:

```suggestion
  end
  
  def fail_and_exit(message)
    Kafo::Helpers.fail_and_exit(message)
  end
```

I'm also tempted to extract the helpers cleanup to a separate commit.

Reply:
I think the behavior here preserves the way it was working the best and co-locates the log methods. In general, this will need clean up but I really want to do it *after* this postgres work is merged. I can file a Redmine issue to track it if that helps.

Reply:
Probably a personal preference, but I'd have this PR as 2 commits: one that builds the infrastructure with `fail_and_exit` and the `HookContextExtension` and then one that only does the actual migration. Not a real difference in the end. Just a bit of personal preference in how I develop :)